app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,71 @@
+class CoursesController < ApplicationController
+  layout 'dashboard'
+  before_action :authenticate_user!
+
+  def new_course
+    @page_title = 'Dashboard -> Add a course'
+    @course = Course.new({})
+    authorize @course
+  end
+
+  def create_course
+    @user = current_user
+    @course = @user.courses.new(course_params)
+    authorize @course if @course.present?
+    @enrollment = @course.enrollments.build(user: @user, enrollment_type: "instructor")
+    if @course.save && @enrollment.save
+      flash[:notice] = 'This course has been added'
+      redirect_to dashboard_show_course_path
+    else
+      render :new_course, status: :unprocessable_entity
+    end
+  end
+
+  def show_course
+    @page_title = 'Dashboard -> Show courses'
+    @courses = policy_scope(Course)
+    authorize @courses
+  end
+
+  def show_single_course
+    @page_title = 'Dashboard -> Show a courses'
+    @course = Course.includes(:lessons).find_by(id: params[:id])
+    authorize @course
+    @lesson = Lesson.new({})
+    @content = Content.new({})
```

Comment:
ditto

Reply:
I remove it in another branch.
Thank you.